File: tests/test-java16/src/main/java/org/example/records/BaseModel.java
package org.example.records;

import io.ebean.Model;
import io.ebean.annotation.Identity;

import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import jakarta.persistence.Version;

@Identity(start = 1000, cache = 100)
@MappedSuperclass
public class BaseModel extends Model {

  @Id
  long id;

  @Version
  long version;

  public long id() {
    return id;
  }

  public void id(long id) {
    this.id = id;
  }

  public long version() {
    return version;
  }

  public void version(long version) {
    this.version = version;
  }
}

File: tests/test-kotlin/src/main/kotlin/org/example/domain/BaseModel.kt
package org.example.domain;

import java.time.Instant;

import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import jakarta.persistence.Version;

import io.ebean.Model;
import io.ebean.annotation.WhenCreated;
import io.ebean.annotation.WhenModified;

@MappedSuperclass
abstract class BaseModel : Model() {
  @Id
  val id: Long? = null

  @Version
  val version: Int = 0

  @WhenCreated
  val whenCreated: Instant? = null

  @WhenModified
  val whenModified: Instant? = null
}
